tests/gen_code.c

```diff
@@ -0,0 +1,301 @@
+{
+	if (!fd)
+		return;
+
+	uri = strdup(uri_in);
+	if (lang == C_LANG) {
+		fprintf(fd, "\t/* Create IIO Context */\n"
+		    "\tIIO_ASSERT(ctx = iio_create_context_from_uri(\"%s\"));\n\n", uri);
+	} else if (lang == PYTHON_LANG) {
+		fprintf(fd, "    # Create IIO Context\n"
+			    "    try:\n"
+			    "        ctx = iio.Context(\"%s\")\n", uri);
+		fprintf(fd, "    except OSError as e:\n"
+			    "        print(\"Unable to open context %s\")\n", uri);
+		fprintf(fd, "        exit(1)\n\n");
+	}
+}
+
+void gen_context_destroy()
+{
+	if (!fd)
+		return;
+
+	if (lang == C_LANG) {
+		fprintf(fd, "\n\t/* Close context at %s, can release/destroy things */\n", uri);
+		fprintf(fd, "\tiio_context_destroy(ctx);\n\treturn EXIT_SUCCESS;\n}\n");
+	} else if (lang == PYTHON_LANG) {
+		/* No need to close context in python
+		 * destroy is called when the destructor is called for iio objects
+		 */
+		fprintf(fd, "\n\nif __name__ == \"__main__\":\n"
+			    "    main()\n");
+	}
+
+	fclose(fd);
+	free(uri);
+}
+
+void gen_context_attr(const char *key)
+{
+	if (!fd)
+		return;
+
+	if (lang == C_LANG) {
+		fprintf(fd, "\t/* Read IIO Context attribute and return result as string */\n");
+		fprintf(fd, "\tval_str = iio_context_get_attr_value(ctx, \"%s\");\n", key);
+		fprintf(fd, "\tprintf(\"%s : %%s\\n\", val_str);\n", key);
+	} else if (lang == PYTHON_LANG) {
+		fprintf(fd, "    # Read IIO Context attribute and return result as string\n");
+		fprintf(fd, "    print(\"%s : \" + ctx.attrs[\"%s\"])\n", key, key);
+	}
+}
+
+void gen_dev(const struct iio_device *dev)
+{
+	if (!fd)
+		return;
+	if (lang == C_LANG) {
+		fprintf(fd, "\t/* Find IIO device in current context */\n");
+		fprintf(fd, "\tIIO_ASSERT(dev = iio_context_find_device(ctx, \"%s\"));\n\n",iio_device_get_name(dev));
+	} else if (lang == PYTHON_LANG) {
+		fprintf(fd, "    # Find IIO device in current context\n");
+		fprintf(fd, "    dev = ctx.find_device(\"%s\")\n\n",iio_device_get_name(dev));
+	}
+}
+
+void gen_ch(const struct iio_channel *ch)
+{
+	const char *name;
+
+	if (!fd)
+		return;
+
+	name = iio_channel_get_name(ch);
+	if (!name)
+		name = iio_channel_get_id(ch);
+
+	if (lang == C_LANG) {
+		fprintf(fd, "\t/* Find the IIO %s channel in the current device */\n",
+			iio_channel_is_output(ch) ? "Output" : "Input");
+		fprintf(fd, "\tIIO_ASSERT(ch = iio_device_find_channel(dev, \"%s\", %s));\n\n",
+			name, iio_channel_is_output(ch) ? "true" : "false");
+	} else if (lang == PYTHON_LANG) {
+		fprintf(fd, "    #Find the IIO %s channel in the current device */\n",
+			iio_channel_is_output(ch) ? "Output" : "Input");
+		fprintf(fd, "    ch = dev.find_channel('%s', %s)\n",
+			name, iio_channel_is_output(ch) ? "True" : "False");
+	}
+}
+
+void gen_context_timeout(unsigned int timeout_ms)
+{
+	if (!fd)
+		return;
+
+	if (lang == C_LANG) {
+		fprintf(fd, "\t/* Set the context timeout in ms */\n");
+		fprintf(fd, "\tiio_context_set_timeout(ctx, %i);\n", timeout_ms);
```

Comment:
static analysis complains about %i. It should be replaced with %u as timeout_ms is unsigned.

Reply:
sorry - should have looked at that - fixed now.